app/javascript/mastodon/rtl.js

```diff
@@ -23,5 +23,14 @@ export function isRtl(text) {
     return false;
   }
 
+  // Remove mentions before counting characters to decide RTL ratio
+  text = text.replace(/@[0-9A-Za-z_@]+/g, '');
```

Comment:
Well, this detects it in RT while you type at the compose window. I don't know the language detection algo you mentioned, because These two are the RTL detection functions I have found. I only extended the existing code, didn't invent new design.

Also, I have no idea how the language detection work (I'll go read it) but right now my tests show it's far from perfect. I found it masked legit toots from the public timelne, I was testing only English and Hebrew, and marked those two languages in, but maybe half the toots were displayed, so it would be a step back from this algo.